lib/compress/zstd_compress.c

```diff
@@ -2603,6 +2590,20 @@ static size_t ZSTD_compressContinue_internal (ZSTD_CCtx* cctx,
     if (cctx->appliedParams.ldmParams.enableLdm)
         ZSTD_window_update(&cctx->ldmState.window, src, srcSize);
 
+    if (ZSTD_window_needOverflowCorrection(ms->window, (const char*)src + srcSize)) {
```

Comment:
I think we will need to duplicate this logic in both here or `ZSTD_compressBlock()` and in the loop in `ZSTD_compress_frameChunk()`.

I'd rather put it in `ZSTD_compressBlock()`, since I don't think the logic will work as expected when `srcSize` is very large.